int32 class_type
float64[3] position
float64[9] rotation